velox/type/TimestampConversion.cpp

```diff
@@ -643,7 +744,7 @@ Expected<int32_t> fromDateString(const char* str, size_t len, ParseMode mode) {
   return daysSinceEpoch;
 }
 
-int32_t extractISODayOfTheWeek(int32_t daysSinceEpoch) {
+int32_t extractISODayOfTheWeek(int64_t daysSinceEpoch) {
```

Comment:
Wondering if we have unit test to show this change is necessary. Thanks.

Reply:
I added a unit test in 0e9412c484304aafe92fa12211ead7425db9bc23. In the previous implementation, the daysSinceEpoch argument could overflow, resulting in an incorrect return value.

Reply:
@NEUpanning Any chance this change can be extracted into a separate PR? It is a bit hard to see it among other changes here.